test/pg_regress.sh

```diff
@@ -102,19 +102,32 @@ fi
 
 function cleanup() {
   rm -rf ${EXE_DIR}/sql/dump
-  rm -rf ${TEST_TABLESPACE1_PATH}
-  rm -rf ${TEST_TABLESPACE2_PATH}
-  rm -rf ${TEST_TABLESPACE3_PATH}
+  rm -rf ${TEST_TABLESPACE1_PREFIX}
+  rm -rf ${TEST_TABLESPACE2_PREFIX}
+  rm -rf ${TEST_TABLESPACE3_PREFIX}
   rm -f ${TEMP_SCHEDULE}
   rm -rf ${TEST_OUTPUT_DIR}/.pg_init
 }
 
 trap cleanup EXIT
 
+# Generating a prefix directory for all test tablespaces. This should
+# be used to build a full path for the tablespace. Note that we
+# terminate the prefix with the directory separator so that we can
+# easily generate paths independent of the OS.
+#
 # This mktemp line will work on both OSX and GNU systems
-TEST_TABLESPACE1_PATH=${TEST_TABLESPACE1_PATH:-$(mktemp -d 2>/dev/null || mktemp -d -t 'timescaledb_regress')}
-TEST_TABLESPACE2_PATH=${TEST_TABLESPACE2_PATH:-$(mktemp -d 2>/dev/null || mktemp -d -t 'timescaledb_regress')}
-TEST_TABLESPACE3_PATH=${TEST_TABLESPACE3_PATH:-$(mktemp -d 2>/dev/null || mktemp -d -t 'timescaledb_regress')}
+TEST_TABLESPACE1_PREFIX=${TEST_TABLESPACE1_PREFIX:-$(mktemp -d 2>/dev/null || mktemp -d -t 'timescaledb_regress')/}
```

Comment:
Personally I would try to avoid using any randomization in testing. Maybe it is possible to create tablespaces based on currently used database name instead?

Reply:
Yes, it is, but the code was there before and has worked fine, so I would argue that there is no reason to change it.

Reply:
FWIW, this is the location on disk of the directory to create the data in. I think it is appropriate to use `mktemp` since that takes care of platform issues. Using the database name doesn't solve the issue of where to create the directory on the system.

Reply:
In theory, you could create a fixed directory on Linux and put the data there instead of using `mktemp`. This is what Windows does. However, I do not think that is a good approach since that means that you cannot run multiple test sessions on the same machine, which is a pointless restriction when it can be avoided by using `mktemp`.

Reply:
if we really wanted to support multiple sessions then we would need to make the port dynamic too since thats what currently prevents running multiple test suites

Reply:
This commit only fixes that objects spanning several databases are made unique. There is more work to do here, but I suggest to do that in separate PRs.

Reply:
yes i wasn't suggesting to fix it in this PR